File: 알고리즘/정렬/selection_sort.py
import random

def selection_sort(data):

    for i in range(len(data) - 1): 
        target: int = i
        for j in range(i + 1, len(data)):
            if data[target] > data[j]:
                target = j

        if target != i:
            data[i], data[target] = data[target], data[i]

    return data

data_list: list = random.sample(range(100), 50)

print(selection_sort(data_list))

File: 알고리즘/정렬/selection_sort.js
function selectionSort() {
  const array = [80, 30, 90, 40, 50, 70, 60, 20, 10, 100, 50];

  for (let i = 0; i < array.length; i++) {
    let minNum = array[i];
    let minIdx = i;
    for (let j = i + 1; j < array.length; j++) {
      if (minNum > array[j]) {
        minNum = array[j];
        minIdx = j;
      }
    }
    [array[i], array[minIdx]] = [array[minIdx], array[i]];
  }
  return array;
}

console.log(selectionSort());
